src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -105,6 +105,121 @@
                     </include>
                 </foreach>
             </if>
+            <if test="studyViewFilter.mutationDataFilters != null and !studyViewFilter.mutationDataFilters.isEmpty()">
+                <foreach item="mutationDataFilter" collection="studyViewFilter.mutationDataFilters" open="INTERSECT" separator="INTERSECT">
+                    <!-- if the categorization is 'MUTATED' (pie chart) -->
+                    <if test="mutationDataFilter.categorization == @org.cbioportal.web.parameter.MutationOption@MUTATED">
+                        WITH total_samples AS (
+                            SELECT sample_unique_id
+                            FROM sample_derived
+                            WHERE cancer_study_identifier IN
+                                <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                                    #{studyId}
+                                </foreach>
+                        ),
+                        profiled_samples AS (
+                            SELECT DISTINCT sgp.sample_unique_id
+                            FROM sample_to_gene_panel_derived sgp
+                            JOIN gene_panel_to_gene_derived gpg ON sgp.gene_panel_id = gpg.gene_panel_id
+                            WHERE cancer_study_identifier IN
+                                <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                                    #{studyId}
+                                </foreach>
+                                AND gpg.gene = #{mutationDataFilter.hugoGeneSymbol}
+                        ),
+                        mutated_samples AS (
+                            SELECT DISTINCT sample_unique_id
+                            FROM genomic_event_derived
+                            WHERE cancer_study_identifier IN
+                                <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                                    #{studyId}
+                                </foreach>
+                                AND hugo_gene_symbol = #{mutationDataFilter.hugoGeneSymbol}
+                                AND variant_type = 'mutation'
+                        )
+                        SELECT DISTINCT sample_unique_id
+                        FROM (
+                            <foreach item="dataFilterValue" collection="mutationDataFilter.values[0]" separator="UNION ALL">
+                                <if test="dataFilterValue.value == 'MUTATED'">
```

Comment:
i think <choose> might be a better option than serial <if>

Reply:
What do you mean

Reply:
Ok I'll change it